tst/EnergyPlus/unit/ZoneEquipmentManager.unit.cc

```diff
@@ -4406,3 +4410,78 @@ TEST_F(EnergyPlusFixture, ZoneAirMassFlowBalance_ZoneMixingInfiltrationFlowsFlag
+
+    state->dataHeatBalFanSys->MAT(1) = 21.0;
+    state->dataHeatBalFanSys->MAT(2) = 22.0;
+    state->dataHeatBalFanSys->ZoneAirHumRat(1) = 0.0021;
+    state->dataHeatBalFanSys->ZoneAirHumRat(2) = 0.0022;
+
+    state->dataHeatBal->TotRefDoorMixing = 1;
+    state->dataHeatBal->TotMixing = 0;
+    state->dataHeatBal->TotCrossMixing = 0;
+    state->dataHeatBal->TotInfiltration = 0;
+    state->dataHeatBal->TotZoneAirBalance = 0;
+    state->dataHeatBal->TotVentilation = 0;
+    state->dataHeatBal->AirFlowFlag = true;
+
+    state->dataContaminantBalance->Contaminant.CO2Simulation = true;
+    state->dataContaminantBalance->Contaminant.GenericContamSimulation = true;
+    state->dataContaminantBalance->MixingMassFlowCO2.allocate(state->dataGlobal->NumOfZones);
+    state->dataContaminantBalance->MixingMassFlowGC.allocate(state->dataGlobal->NumOfZones);
+
+    state->dataHeatBal->RefDoorMixing.allocate(state->dataGlobal->NumOfZones);
+    state->dataHeatBal->RefDoorMixing(1).EMSRefDoorMixingOn.allocate(state->dataGlobal->NumOfZones);
+    state->dataHeatBal->RefDoorMixing(1).VolRefDoorFlowRate.allocate(state->dataGlobal->NumOfZones);
+    state->dataHeatBal->RefDoorMixing(1).MateZonePtr.allocate(state->dataGlobal->NumOfZones);
+    state->dataHeatBal->RefDoorMixing(1).RefDoorMixFlag = true;
+    state->dataHeatBal->RefDoorMixing(2).RefDoorMixFlag = true;
+    state->dataHeatBal->RefDoorMixing(1).NumRefDoorConnections = 1;
+    state->dataHeatBal->RefDoorMixing(1).MateZonePtr(1) = 2;
+    state->dataHeatBal->RefDoorMixing(1).EMSRefDoorMixingOn(1) = true;
+    state->dataHeatBal->RefDoorMixing(1).VolRefDoorFlowRate(1) = 100.0;
+
+    state->dataContaminantBalance->ZoneAirCO2.allocate(state->dataGlobal->NumOfZones);
+    state->dataContaminantBalance->ZoneAirGC.allocate(state->dataGlobal->NumOfZones);
+    state->dataContaminantBalance->ZoneAirCO2(1) = 400.0;
+    state->dataContaminantBalance->ZoneAirCO2(2) = 300.0;
+    state->dataContaminantBalance->ZoneAirGC(1) = 10.0;
+    state->dataContaminantBalance->ZoneAirGC(2) = 20.0;
+
+    state->dataEarthTube->GetInputFlag = false;
+    state->dataEarthTube->TotEarthTube = 0;
+    state->dataCoolTower->GetInputFlag = false;
+    state->dataCoolTower->NumCoolTowers = 0;
+    state->dataThermalChimneys->ThermalChimneyGetInputFlag = false;
+    state->dataThermalChimneys->TotThermalChimney = 0;
+
+    state->dataHeatBal->RefDoorMixing(1).VolRefDoorFlowRate(1) = 1.1;
+    state->dataHeatBal->RefDoorMixing(1).VolRefDoorFlowRate(2) = 2.2;
+
+    CalcAirFlowSimple(*state);
+
+    EXPECT_NEAR(state->dataContaminantBalance->MixingMassFlowCO2(1), 395.202, 0.001);
+    EXPECT_NEAR(state->dataContaminantBalance->MixingMassFlowCO2(2), 526.884, 0.001);
+    EXPECT_NEAR(state->dataContaminantBalance->MixingMassFlowGC(1), 26.347, 0.001);
+    EXPECT_NEAR(state->dataContaminantBalance->MixingMassFlowGC(2), 13.172, 0.001);
+}
```

Comment:
Nice unit test!

I think it looks pretty good. I am going run some comparisons with a local pull and branch updates.